examples/chef/common/clusters/resource-monitoring/chef-resource-monitoring-delegates.cpp

```diff
@@ -96,25 +98,161 @@ void HepaFilterMonitoring::Shutdown()
+    case HepaFilterMonitoring::Id:
+        ret = gHepaFilterInstance->ExternalAttributeWrite(attributeMetadata, buffer);
+        break;
+    case ActivatedCarbonFilterMonitoring::Id:
+        ret = gActivatedCarbonFilterInstance->ExternalAttributeWrite(attributeMetadata, buffer);
+        break;
+    default:
+        ret = Protocols::InteractionModel::Status::UnsupportedWrite;
+        break;
+    }
+
+    return ret;
+}
+
+chip::Protocols::InteractionModel::Status
+chefResourceMonitoringExternalReadCallback(chip::EndpointId endpoint, chip::ClusterId clusterId,
+                                           const EmberAfAttributeMetadata * attributeMetadata, uint8_t * buffer,
+                                           uint16_t maxReadLength)
+{
+    Protocols::InteractionModel::Status ret = Protocols::InteractionModel::Status::Success;
+    AttributeId attributeId                 = attributeMetadata->attributeId;
+    ChipLogProgress(Zcl, "chefResourceMonitoringExternalReadCallback EP: %d, Cluster: %d, Att: %d", static_cast<int>(endpoint),
+                    static_cast<int>(clusterId), static_cast<int>(attributeId));
+
+    switch (clusterId)
+    {
+    case HepaFilterMonitoring::Id:
+        ret = gHepaFilterInstance->ExternalAttributeRead(attributeMetadata, buffer, maxReadLength);
+        break;
+    case ActivatedCarbonFilterMonitoring::Id:
+        ret = gActivatedCarbonFilterInstance->ExternalAttributeRead(attributeMetadata, buffer, maxReadLength);
+        break;
+    default:
+        ret = Protocols::InteractionModel::Status::UnsupportedRead;
+        break;
+    }
+
+    return ret;
+}
+
+chip::Protocols::InteractionModel::Status
+ChefResourceMonitorInstance::ExternalAttributeRead(const EmberAfAttributeMetadata * attributeMetadata, uint8_t * buffer,
+                                                   uint16_t maxReadLength)
+{
+    Protocols::InteractionModel::Status ret = Protocols::InteractionModel::Status::Success;
+    AttributeId attributeId                 = attributeMetadata->attributeId;
+
+    switch (attributeId)
+    {
+    case HepaFilterMonitoring::Attributes::Condition::Id: {
+        *buffer = GetCondition();
+    }
+    break;
+    case HepaFilterMonitoring::Attributes::ChangeIndication::Id: {
+        ResourceMonitoring::ChangeIndicationEnum changeIndication = GetChangeIndication();
+        // The underlying type of ResourceMonitoring::ChangeIndicationEnum is uint8_t
+        *buffer = to_underlying(changeIndication);
+    }
+    break;
+    case HepaFilterMonitoring::Attributes::InPlaceIndicator::Id: {
+        *(bool *) buffer = GetInPlaceIndicator();
+    }
+    break;
+    case HepaFilterMonitoring::Attributes::LastChangedTime::Id: {
+        DataModel::Nullable<uint32_t> lastChangedTime = GetLastChangedTime();
+        *(uint32_t *) buffer                          = lastChangedTime.IsNull() ? 0 : lastChangedTime.Value();
+    }
+    break;
+    case HepaFilterMonitoring::Attributes::DegradationDirection::Id:
```

Comment:
If the condition attribute is supported shouldn't the degradation direction also be supported? They both fall under the CON feature for conformance.

Reply:
@harimau-qirex Actually the limitation comes from here https://source.corp.google.com/search?q=%22%2Fcodelabs%2Fmatter-device-virtual%22&sq=package:piper%20file:%2F%2Fdepot%2Fgoogle3%20-file:google3%2Fexperimental . We just support everything except `degradation direction`

```
    Protocols::InteractionModel::Status UpdateCondition(uint8_t aNewCondition);
    Protocols::InteractionModel::Status UpdateChangeIndication(ChangeIndicationEnum aNewChangeIndication);
    Protocols::InteractionModel::Status UpdateInPlaceIndicator(bool aNewInPlaceIndicator);
    Protocols::InteractionModel::Status UpdateLastChangedTime(DataModel::Nullable<uint32_t> aNewLastChangedTime);
```

Reply:
Per discussion above, this is the current cluster implementation limitation. Mark this as resolved.

Reply:
This is reading the attribute, not writing the attribute. The corresponding cluster implementation method is https://github.com/project-chip/connectedhomeip/blob/6842912989bd43a0b15e069eaea33dfc5495de04/src/app/clusters/resource-monitoring-server/resource-monitoring-server.h#L104.

Reply:
OK. The reading should be supported. Thanks for catching this. The fix is done and committed. So I'll mark this as resolved.